src/components/MultiHopTrade/components/TradeInput/hooks/useTradeReceiveAddress.tsx

```diff
@@ -0,0 +1,20 @@
+import { useReceiveAddress } from 'components/MultiHopTrade/hooks/useReceiveAddress'
+import {
+  selectInputBuyAsset,
+  selectLastHopBuyAccountId,
+  selectManualReceiveAddress,
+} from 'state/slices/selectors'
+import { useAppSelector } from 'state/store'
+
+export const useTradeReceiveAddress = () => {
+  const buyAsset = useAppSelector(selectInputBuyAsset)
+  const buyAccountId = useAppSelector(selectLastHopBuyAccountId)
+  const manualReceiveAddress = useAppSelector(selectManualReceiveAddress)
+
+  const { walletReceiveAddress, isLoading } = useReceiveAddress({
```

Comment:
suggestion: `useReceiveAddress` is actually only used in the context of trade (spot/limit) which can be confusing as it reads like a general-purpose hook.
What about something along the lines of `useSwapperReceiveAddress` ?

Reply:
Ohh, sounds sane to me!